core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -397,7 +398,8 @@ protected void startServingRPCServer() {
         .processor(processor)
         .transportFactory(transportFactory)
         .protocolFactory(ThriftUtils.createThriftProtocolFactory())
-        .stopTimeoutVal((int) Configuration.getMs(PropertyKey.MASTER_THRIFT_SHUTDOWN_TIMEOUT));
+        .stopTimeoutVal((int) TimeUnit.MILLISECONDS
```

Comment:
Should we also explicitly set `stopTimeoutUnit` to be `TimeUnit.SECONDS`?

Reply:
@gpang Yeah I thought about that. There is no method in `TThreadPoolServer.Args` to set that :-(

Reply:
@madanadit Should we just set it directly, since it is `public`?

Reply:
yes, didn't realize it was public

Reply:
@gpang The thrift method `TThreadPoolServer.createDefaultExecutorService` is strange. It does not respect the time unit (if not set to seconds). So we must set to the unit to seconds 
```
  private static ExecutorService createDefaultExecutorService(Args args) {
    SynchronousQueue<Runnable> executorQueue =
      new SynchronousQueue<Runnable>();
    return new ThreadPoolExecutor(args.minWorkerThreads,
                                  args.maxWorkerThreads,
                                  args.stopTimeoutVal,
                                  TimeUnit.SECONDS,
                                  executorQueue);
  }```